agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -1298,3 +1328,35 @@ func (dg *dockerGoClient) loadImage(ctx context.Context, reader io.Reader) error
 	_, err = client.ImageLoad(ctx, reader, false)
 	return err
 }
+
+type proxyReader struct {
+	io.ReadCloser
+	calls uint64
+}
+
+func (p *proxyReader) callCount() uint64 {
+	return atomic.LoadUint64(&p.calls)
+}
+
```

Comment:
why is this method omitted: https://github.com/fsouza/go-dockerclient/blob/a7b85d61dcf384ef9fea13abaa9fefe5ef00ad25/client.go#L663 ? i think you need it in order for inactivity timeout to work